PowerEditor/src/MISC/PluginsManager/Notepad_plus_msgs.h

```diff
@@ -549,6 +549,71 @@ enum Platform { PF_UNKNOWN, PF_X86, PF_X64, PF_IA64, PF_ARM64 };
 	// void* NPPM_GETBOOKMARKID(0, 0)
 	// Returns the bookmark ID
 
+	#define NPPM_DARKMODESUBCLASSANDTHEME (NPPMSG + 112)
+	// ULONG NPPM_DARKMODESUBCLASSANDTHEME(ULONG dmFlags, HWND hwnd)
+	// Subclass hwnd to add support for generic dark mode.
+	// When applying first time on main hwnd use "dmFlags = NppDarkMode::dmfAll".
+	// In case edit, listbox, static text, treeview, listview and toolbar controls don't have parent as main hwnd,
+	// call NPPM_DARKMODESUBCLASSANDTHEME on their parent and use "dmFlags = NppDarkMode::dmfSetParent".
+	// When handling dark mode change with NPPN_DARKMODECHANGED, use "dmFlags = NppDarkMode::dmfSetChildren".
+	// 
+	// Docking panels don't need to call NPPM_DARKMODESUBCLASSANDTHEME for main hwnd.
+	// Subclassing is applied automatically unless DWS_USEOWNDARKMODE flag is used.
+	//
+	// Might not work properly in C# plugins.
+	//
+	// Returns succesful combinations from dmfRequiredMask.
+	//
+
+	namespace NppDarkMode
+	{
+		// used on parent of edit, listbox, static text, treeview, listview and toolbar controls
+		constexpr ULONG dmfSetParent =          0x00000001UL;
+		// without dmfSubclassChildren will only theme
+		// should be handled on first time and in NPPN_DARKMODECHANGED
+		// only theme requires at least Windows 10
+		constexpr ULONG dmfSetChildren =        0x00000002UL;
+		// should be used only on main hwnd when applying subclass first time
+		constexpr ULONG dmfSubclassChildren =   0x00000004UL;
+		// set dark title bar
+		// should be handled on first time and in NPPN_DARKMODECHANGED
+		// requires at least Windows 10
+		constexpr ULONG dmfSetTitleBar =        0x00000008UL;
+		// set dark explorer theme
+		// used mainly for dark scrollbar not handled with dmfSetChildren
+		// might change style for other elements
+		// should be handled on first time and in NPPN_DARKMODECHANGED
+		// requires at least Windows 10
+		constexpr ULONG dmfSetExplorerTheme =   0x00000010UL;
+		// standard first time flags for main parent
+		constexpr ULONG dmfMainParent = dmfSetParent | dmfSetChildren | dmfSubclassChildren | dmfSetTitleBar; // 0x000000FUL
+		// at least one of these flag should be used
+		constexpr ULONG dmfRequiredMask = dmfSetParent | dmfSetChildren | dmfSetTitleBar | dmfSetExplorerTheme; // 0x0000001BUL
+		// every flags
+		constexpr ULONG dmfAllMask = dmfSetParent | dmfSetChildren | dmfSubclassChildren | dmfSetTitleBar | dmfSetExplorerTheme; // 0x0000001FUL
+	};
+
+	// Examples:
+	//
+	// - first time:
+	//
+	//auto success = static_cast<ULONG>(::SendMessage(nppData._nppHandle, NPPM_DARKMODESUBCLASSANDTHEME, static_cast<WPARAM>(NppDarkMode::dmfMainParent), reinterpret_cast<LPARAM>(mainHwnd)));
+	//
+	// - handling dark mode change:
+	//
+	//extern "C" __declspec(dllexport) void beNotified(SCNotification * notifyCode)
+	//{
+	//	switch (notifyCode->nmhdr.code)
+	//	{
+	//		case NPPN_DARKMODECHANGED:
+	//		{
+	//			constexpr auto flags = NppDarkMode::dmfSetChildren | NppDarkMode::dmfSetTitleBar;

```

Comment:
The settings of flags is quite confusing: in which case plugin authors should use which flag?

Reply:
Again same approach as Notepad++ dark mode change handling.

`dmfSetChildren` is required for mainly push buttons which use theme rather than subclassing.
 Title bars also don't use subclassing. If dialog has title bar it should use `dmfSetTitleBar`.

Maybe better explanation in comments or documentation should help.